### PR TITLE
Add wall lines to landuse layer

### PR DESCRIPTION
**Pull request by nvkelso**
Connects with #1403 to add wall lines to landuse layer

- [x] Update tests
- [x] Update data migrations
- [x] Update docs

### CODE REVIEW

File: yaml/landuse.yaml [hunk 1]
```diff
@@ -1030,11 +1030,18 @@ filters:
     output:
       <<: *output_properties
       kind: retaining_wall
+  - filter: {barrier: wall}
+    min_zoom: 16
+    output:
+      <<: *output_properties
+      kind: wall
+      kind_detail: { col: wall }
```
Comment (zerebubuth) [inline yaml/landuse.yaml]: Is it worth whitelisting the most common values for the `wall` tag value?

Comment (nvkelso) [reply yaml/landuse.yaml]: From https://taginfo.openstreetmap.org/keys/wall#values:

wall_value | count
---------- | -----
no | 11704961
dry_stone | 92635
noise_barrier | 7787
brick | 7282
yes | 5209
stone | 2311
pise | 1829
castle_wall | 891
null | 575
jersey_barrier | 566
seawall | 492
flood_wall | 420
drystone | 357
concrete | 292
retaining_wall | 238
stone_bank | 211
gabion | 181
D | 154
wind_barrier | 130
kő | 126
alvenaria | 113
wood | 101
kőfal | 89

Comment (nvkelso) [reply yaml/landuse.yaml]: The general list summary (see table above) is already added to documentation – we're split 50/50 on when to just pass thru and document common values versus white listing. 

I can see black listing a few types like:

wall_value | count
-- | --
no | 11704961
yes | 5209
null | 575

Comment (nvkelso) [reply yaml/landuse.yaml]: I'm not seeing many combinations of `barrier=wall` and `wall=[no,yes,null]` in the wild so I'm going to merge this as-is. If the problem exists in the wild we can revisit later.

**Comment by zerebubuth** [comment]
> we're split 50/50 on when to just pass thru and document common values versus white listing

As a future task, I'd like us to move towards white-listing by default, with rare, deliberate and documented exceptions for things such as `name`. I think the benefit is to cartographers and map publishers who are able to know in advance from the documentation what they have to cover (e.g: icons, `filter` clauses) to display the features on the map.

This would mean that we should regularly review the white-lists to ensure they stay up to date with the tagging, and perhaps this is something we need tooling around.

**Pull request merged**